### DISCUSSION TITLE
Integração dos Testes

### DISCUSSION THREAD

**Discussion by moisesAlc** [falae-it-talent/falae#18] [General]
Pessoal, o Siqueira está iniciando uma discussão da integração da parte de testes. 

Ele criou uma "massa de testes": informações no banco de dados e usuários. É necessário passar esses dados para o novo ambiente que virá. Quais etapas devemos fazer?

**Comment by lucasarags** [comment]
Se é um ambiente de teste com o banco de dados, seria interesse ter um banco separado do de prod so para esse tipo de teste. Se fosse um cenário real, teriamos nossa aplicação rodando em um ambiente de prod. Como ele quer criar uma simulação do ambiente em uso, teria que termos esse ambiente de teste desenvolvido para utilizar em nuvem. Nosso projeto não foi definido ambientes de uso, pelo que entendi desde o começo, seria tudo em um único ambiente.

Neste caso existe sim uma solução para esse problema, por temos duas instâncias disponíveis, talvez seria interesse pensarmos em dois tipos de ambientes. Um para prod, na instancia maior, e outro somente para teste em dev e QA. assim não vamos misturar dados de um cenário real em uso com o cenário simulado em uso.

**Reply by lucasarags** [reply]
Além disso, em um futuro próximo, poderiamos utlizar o código do terrqaform para gerar uma pipeline de criação e destruição da instancia, com foco redução de gastos, mas isso é conversa para outra discussão. são muitas ideias e sugestões que podemos fazer dentro do falae para melhoria do desenvolvimento.

**Comment by lucasarags** [comment]
Poderiamos inclusive inserir esse cenário de teste com esses dados simulados, juntamente com outros testes de qa automaizados, tudo sendo feito com a pipeline para testes automatizados de qa. Este é o grande desafio para este projeto, refetente a pipeline

**Reply by lucasarags** [reply]
Outra solução simples e prática, seria ele já usar a instância que ele tem disponivel e jogar no banco estes usuráios e dados. tendo assim o cenário dele.
